database/migrations/2021_02_10_100000_add_checkout_permissions.php

```diff
@@ -13,5 +13,19 @@ public function up()
             'view order items',
         ];
         $this->createPermissions($permissions);
+
+        $permissionsByRole = [
+            'Owner' => [
+                'view orders',
+                'view order items',
+            ],
+            'Staff' => [
+                'view orders',
+                'view order items',
+            ],
+        ];
+        foreach ($permissionsByRole as $role => $permissions) {
+            $this->addPermissionsToRole($permissions, $role);
+        }
```

Comment:
I like the way this looks. Can the foreach loop be further simplified in the tipoff/support function?

It would be nice to get it to where there is just a function that gets passed the array or roles & permissions.

Reply:
@pdbreen Do you have a suggestion for this? Take a look at this referenced PR:

- https://github.com/tipoff/authorization/pull/41

Reply:
You could change the first array so every permission is followed by the its roles.  This ensures all permissions are always created and granted to admin regardless of whether they are granted to some other role.
```
$permissions = [
   'view orders' => ['Owner', 'Staff'],
   'view order items'  => ['Owner', 'Staff'],
  'some admin only permission' => [],
];
```

Reply:
I like that approach. Thank you!

Reply:
Looks great, should we just implement this in createPermissions instead of having a separate function?

Reply:
For admin-only permissions, why not spedify 'Admin' in the array, ie.
```
$permissions = [
    'view orders' => ['Owner', 'Staff'],
    'view order items'  => ['Owner', 'Staff'],
    'admin only permission' => ['Admin'],
];

Reply:
`Admin` role should always get every permission regardless of whether admin role is listed.  Having it listed sometimes might be confusing.

Reply:
I agree. Admin will get every permission by default and we'll just add the other roles to that array.

Yes, let's just modify that createPermissions function and remove the other function so there is only one new version.

Reply:
Done. Thank you @djamesfar 

- https://github.com/tipoff/authorization/pull/43